dms/models/directory.py

```diff
@@ -622,15 +618,23 @@ def _process_message(self, msg_dict, extra_values=False):
     @api.model_create_multi
     def create(self, vals_list):
         for vals in vals_list:
-            if vals.get("root_storage_id", False):
-                vals["storage_id"] = vals["root_storage_id"]
             if vals.get("parent_id", False):
                 parent = self.browse([vals["parent_id"]])
                 data = next(iter(parent.sudo().read(["storage_id"])), {})
                 vals["storage_id"] = self._convert_to_write(data).get("storage_id")
         return super().create(vals_list)
 
     def write(self, vals):
+        if "storage_id" in vals and vals["storage_id"]:
+            storage = self.env["dms.storage"].browse(vals["storage_id"])
+            for item in self:
+                if item.storage_id == storage:
+                    continue
+                if item.storage_id.save_type != storage.save_type:
```

Comment:
I think we should forbid changing the storage at all, no matter the `save_type`, as we don't want to maintain the storage propagation code. Or there is a use case for allowing to change the storage?

Reply:
I think that if you should allow changing the storage because for example you want to re-organize the folders (as long as the storage type is the same).

Reply:
We should forbid the change of the storage at all, as having the same `save_type` doesn't mean that data are saved in the same place. Imagine a new addon for storing in `Google Drive`. The `save_type` will be the same, but the specific parameters of the storage will redirect to a different account/path. This is also happening with the attachments, changing the model on each specific storage.